### DISCUSSION TITLE
Include a call to an initialization method as part of object construction

### DISCUSSION THREAD

**Discussion by msauper** [dotnet/csharplang#4002] [General] [answered]
I'd like to be able to define a method that will be called as the last step of object construction once all object constructors have been called.  This method would support method overrides.  The method could be defined explicitly or convention.  Explicitly, the method could be identified by an attribute or Interface implementation.

**Comment by Joe4evr** [comment, marked as answer]
So basically you want the "validator", [which the team is already considering to add](https://youtu.be/fIXujTOA6BE?t=2749) (though it's probably gonna miss 9.0).

**Reply by msauper** [reply]
Yes, that is what I would like. Although, my use is to implement standard initialization of objects that depend on information provided by inherited objects.

**Reply by Joe4evr** [reply]
Yes, so for that scenario, you would provide some `protected virtual` method in your object that you call from the validator/"final initializer".
